src/Build/BuildCheck/Infrastructure/ConfigurationProvider.cs

```diff
@@ -264,7 +263,7 @@ internal BuildAnalyzerConfigurationInternal GetMergedConfiguration(string projec
             ruleId: ruleId,
             evaluationAnalysisScope: GetConfigValue(editorConfig, defaultConfig, cfg => cfg.EvaluationAnalysisScope),
             isEnabled: GetConfigValue(editorConfig, defaultConfig, cfg => cfg.IsEnabled),
-            severity: GetConfigValue(editorConfig, defaultConfig, cfg => cfg.Severity));
+            severity: GetSeverityValue(editorConfig, defaultConfig));
```

Comment:
Why `GetConfigValue` cannot be used here? I'm trying to see if we need the special handling method `GetSeverityValue`

Reply:
```
 if (editorConfigValue.Severity != null && editorConfigValue.Severity != BuildAnalyzerResultSeverity.Default)
  {
      resultSeverity = editorConfigValue.Severity;
  }
```
In order to skip the default value and to consider it as `null`.

Reply:
Added the corner case described in the method.